cognee/api/v1/cognify/cognify_v2.py

```diff
@@ -25,7 +31,21 @@
 
 update_status_lock = asyncio.Lock()
 
-async def cognify(datasets: Union[str, list[str]] = None, root_node_id: str = None):
+class PermissionDeniedException(Exception):
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+async def cognify(datasets: Union[str, list[str]] = None, root_node_id: str = None, user_id:str="default_user"):
+    session: AsyncSession = async_session_maker()
```

Comment:
Let's put this login in a function in some module and just use it here.
Something like `def check_permissions(user)` for example.

Reply:
Let's discuss this live